client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx

```diff
@@ -5,24 +5,43 @@ import page from 'page';
 import { ReactElement, useCallback, useState } from 'react';
 import { useDispatch } from 'react-redux';
 import FormRadio from 'calypso/components/forms/form-radio';
+import Pagination from 'calypso/components/pagination';
 import SearchCard from 'calypso/components/search-card';
+import { SITE_CARDS_PER_PAGE } from 'calypso/jetpack-cloud/sections/partner-portal/assign-license-form/constants';
 import { addQueryArgs } from 'calypso/lib/url';
 import { recordTracksEvent } from 'calypso/state/analytics/actions';
 import { errorNotice } from 'calypso/state/notices/actions';
 import useAssignLicenseMutation from 'calypso/state/partner-portal/licenses/hooks/use-assign-license-mutation';
 import './style.scss';
 
-export default function AssignLicenseForm( { sites }: any ): ReactElement {
+function setPage( pageNumber: number ): void {
+	const queryParams = { page: pageNumber };
+	const currentPath = window.location.pathname + window.location.search;
+
+	page( addQueryArgs( queryParams, currentPath ) );
+}
+
+function paginate( arr: Array< any >, currentPage: number ): Array< any > {
+	return (
+		arr
+			// Slices sites list based on pagination settings
+			.slice( SITE_CARDS_PER_PAGE * ( currentPage - 1 ), SITE_CARDS_PER_PAGE * currentPage )
+	);
+}
+
+export default function AssignLicenseForm( { sites, currentPage }: any ): ReactElement {
 	const translate = useTranslate();
 	const dispatch = useDispatch();
-	const [ filter, setFilter ] = useState( false );
+	const [ filter, setFilter ] = useState( null );
```

Comment:
Switched to `null` as default state since the `SearchCard` component would set the value of filter to an empty string when input is cleared and TypeScript complains about boolean and string type comparisons. (`hasFilter` flag)

Reply:
Using an empty string instead of null as a default value should make this even simpler then as the value will always be a string.